diff -ruN a/source/effects/decals.c b/source/effects/decals.c
--- a/source/effects/decals.c	2026-10-04 01:02:43.264973230 +0000
+++ b/source/effects/decals.c	2026-10-04 01:42:10.350685380 +0000
@@ -832,34 +832,8 @@
 			color->green,
 			color->blue));
 
-	__asm
-	{
-		mov		edx, color
-		fld		alpha
-		fld		dword ptr [edx]
-		fld		dword ptr [edx+4]
-		fld		dword ptr [edx+8]
-		fld		scale
-		fmul	st(4), st
-		fmul	st(3), st
-		fmul	st(2), st
-		fmulp	st(1), st
-		fistp	result
-		and		result, 0FFh
-		mov		edx, result
-		fistp	result
-		and		result, 0FFh
-		shl		result, 8
-		or		edx, result
-		fistp	result
-		and		result, 0FFh
-		shl		result, 16
-		or		edx, result
-		fistp	result
-		shl		result, 24
-		or		edx, result
-		mov		result, edx
-	}
+	result = (pixel32)((lrintf(color->blue*scale)&0xFF) | ((lrintf(color->green*scale)&0xFF)<<8) |
+		((lrintf(color->red*scale)&0xFF)<<16) | (lrintf(alpha*scale)<<24));
 
 	return result;
 }
